docs/fix_CONTRIBUTING.md (#1984)
PyTorchLightning/pytorch-lightning: .github/CONTRIBUTING.md

## Main Core Value: One less thing to remember

Simplify the API as much as possible from the user perspective.
 Any additions or improvements should minimize the things the user needs to remember.   

For example: One benefit of the validation_step is that the user doesn't have to remember to set the model to .eval().

>  This helps users avoide all sorts of subtle errors.

Reviewer: ```suggestion
 This helps users avoid all sorts of subtle errors.
```

Note: if your computer does not have multi-GPU nor TPU these tests are skipped.

For convenience, you can also use your own CircleCI building which will be triggered with each commit.

> This is useful if you do not test against all required dependencies versions.

Reviewer: ```suggestion
This is useful if you do not test against all required dependency versions.
```

1. **How can I help/contribute?**

    All help is very welcome - reporting bugs, solving issues and preparing bug fixes. To solve some issues you can start with label [good first issue](https://github.com/PyTorchLightning/pytorch-lightning/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) or chose something close to your domain with label [help wanted](https://github.com/PyTorchLightning/pytorch-lightning/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22). Before you start to implement anything check that the issue description that it is clear and self-assign the task to you (if it is not possible, just comment that you take it and we assign it to you...).

2. **Is there a recommendation for branch names?**
    
    We do not rely on the name convention so far you are working with your own fork. Anyway it would be nice to follow this convention `<type>/<issue-id>_<short-name>` where the types are: `bugfix`, `feature`, `docs`, `tests`, ...

3. **How to rebase my PR?**
    

>     We recommend creating a PR in a separate branch, othen than `master`, especially if you plan on submitting several changes at once and do not want to wait until the first one is resolved (we can work on them in parallel). Update your master with upstream (assuming you have already set [upstream](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/configuring-a-remote-for-a-fork))

Reviewer: ```suggestion
    We recommend creating a PR from a separate branch other than `master`, especially if you plan on submitting several changes at once and do not want to wait until the first one is resolved (we can work on them in parallel). Update your master with upstream (assuming you have already set [upstream](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/configuring-a-remote-for-a-fork))
```
